pyproject.toml

```
@@ -1,10 +1,60 @@
-# NB: This requires the use of pyproject-flake8
-[tool.flake8]
-max-line-length = 99
-extend-ignore = "E266,"
+# SPDX-License-Identifier: Apache-2.0
+# Copyright 2022 The Foundry Visionmongers Ltd
+
+[project]
+name = "openassetio-manager-bal"
+version = "1.0.0a2"
+requires-python = ">=3.7"
+dependencies = [
+    "openassetio == 1.0.0a7"
+]
+
+authors = [
+  { name = "Contributors to the OpenAssetIO project", email = "openassetio-discussion@lists.aswf.io" }
+]
+
+description = """\
+    A contrived "asset management system" for OpenAssetIO integration
+    test cases.
+    """
+keywords = ["openassetio", "manager"]
+classifiers = [
+    "Development Status :: 3 - Alpha",
+    "Intended Audience :: Developers",
+    "License :: OSI Approved :: Apache Software License",
+    "Natural Language :: English",
+    "Operating System :: MacOS",
+    "Operating System :: Microsoft :: Windows",
+    "Operating System :: POSIX :: Linux",
+    "Programming Language :: Python :: 3"
+]
+readme = "README.md"
+
+[project.urls]
+Source = "https://github.com/OpenAssetIO/OpenAssetIO-Manager-BAL"
+Issues = "https://github.com/OpenAssetIO/OpenAssetIO-Manager-BAL/issues"
+
+# Defines a Python entry point that exposes the plugin's package to
+# allow entry point based discovery.
+[project.entry-points."openassetio.manager_plugin"]
+openassetio_manager_bal = "openassetio_manager_bal"
```

[consider] It seems only the value is used, not the key, so we're free to give the key a more meaningful name. E.g.
```suggestion
plugin_package_or_module = "openassetio_manager_bal"
```

A bit late to the party, but as a side-note, we could in future be even more flexible and not prescribe the name of the `plugin` attribute
```suggestion
plugin_class = "openassetio_manager_bal:BasicAssetLibraryPlugin"
```

---

Like the alternate name, but Is there merit in using the same loading mechanism as `OPENASSETIO_PLUGIN_PATH` ie: with the `plugin` attribute, then at least you can chop and change and don't have to worry about the alternate code paths working with one and not the other.

---

We could say: if you give us a module/package then we'll look for a `plugin` attribute, whereas if you give us a class, we'll use that as the plugin.

But it was just a (rather belated) thought after reading through the importlib code.  I don't think we need to do anything now.